crypto/mem.c

```diff
@@ -160,8 +160,7 @@ static int shouldfail(void)
 void ossl_malloc_setup_failures(void)
 {
     const char *cp = getenv("OPENSSL_MALLOC_FAILURES");
-
-    if (cp != NULL && (md_failstring = strdup(cp)) != NULL)
+    if (cp != NULL && (md_failstring = OPENSSL_strdup(cp)) != NULL)
```

Comment:
this change is also problematic, as this creates a memory leak,
that is visible to `./config enable-crypto-mdebug`
but only when test is run this way:
`OPENSSL_MALLOC_FAILURES=0 make test`

Reply:
Wasn't the leak there already? One-off memory leak too.

Reply:
yes, but that memory is reachable by static pointer, so address sanitizer will br silent about it,
but the mdebug does not do inspect all global pointers, if the memory in question is still reachable.

Reply:
we could just replace the use of md_failstring in parseit with an immediate call to getenv("OPENSSL_MALLOC_FAILURES") and get rid of md_failstring entirely.